tests/Unit.Tests/Client/HttpClientWrapperTests.cs

```diff
@@ -86,6 +86,24 @@ public void GetAsync_SearchAndCompareApiException()
             act.Should().Throw<SearchAndCompareApiException>().WithMessage(msg);
         }
 
+        [Test]
+        public void GetAsync_SearchAndCompareApi404()
+        {
+            var ub = new UriBuilder("test");
+            var uri = ub.Uri;
+
+            mockMessageHandler
+                .Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage { StatusCode = HttpStatusCode.NotFound });
+
+            Func<Task> act = async () => await sut.GetAsync(uri);
+
+            act.Should().NotThrow<SearchAndCompareApiException>();
```

Comment:
question: this tests that it fails to fail, but can we also test that it's behaving correctly? I can't see from this what the expected behaviour is. Should it throw a different exception, or no exception, or is there something else we can check to validate the response?

Reply:
What it's testing is that is doesn't throw an exception on a 404 which is the expected behaviour.
This test specifically targets the fixed bug and would have flushed it out if it had previously existed. The happy path tests already exist.

Reply:
I don't think it shows the expected behaviour when there's a 404, just the unwanted behaviour we previously had

Reply:
Ok new test added